### PR TITLE
Feature/enterprise blobber refactor alloc

### PR DESCRIPTION
**Pull request by Jayashsatolia403**
## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by dabasov — APPROVED** [review]
lgtm

File: code/go/0chain.net/smartcontract/storagesc/allocation.go [hunk 1]
```diff
@@ -345,25 +390,35 @@ func (sc *StorageSmartContract) newAllocationRequestInternal(
...
 	}
 	m.tick("create_write_pool")
 
-	if err = sc.createChallengePool(txn, sa, balances, conf); err != nil {
-		logging.Logger.Error("new_allocation_request_failed: error creating challenge pool",
-			zap.String("txn", txn.Hash),
-			zap.Error(err))
-		return "", common.NewError("allocation_creation_failed", err.Error())
+	if !request.IsEnterprise {
+		if err = sc.createChallengePool(txn, alloc, balances, conf); err != nil {
```
Comment (peterlimg) [inline code/go/0chain.net/smartcontract/storagesc/allocation.go:406]: We need activation to switch off the createChallengePool, imagine if both old and new miners received the request with `IsEnterprise=true`, the old one will always create the challenge pool, while the new one will skip.

Comment (peterlimg) [reply code/go/0chain.net/smartcontract/storagesc/allocation.go:406]: nvm, seems fixed in PR https://github.com/0chain/0chain/pull/3357

**Comment by peterlimg** [comment]
lgtm

**Pull request merged**